### PR TITLE
Week 10, Survey tool - Therese Hagelin

### PR DESCRIPTION
**Pull request by ThereseHag**
Link to my deployed project:
https://therese-survey.netlify.app/

My reflections on how this week's project turned out:
The work has proceeded in a good way. I started with creating header and footer components, followed by a basic form structure, and then added mapping, statements and summary. I had some issues when trying to deploy it, but after getting the advice to fix all warnings in the terminal before deploying, it all worked out fine. I like React a lot, even if there is still so very much to learn and a lot of information to process. 

Things I'd like to have clarified or explained in more detail:
I would like to get a better understanding of how to work with checkbox inputs, but will make sure to study that after this weeks project. Also, I'm always interested in getting a better understanding on how to use map method, even if it worked out well this time.

### CODE REVIEW

File: code/src/Components/Form.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+const Form = () => {
+  const [genre, setGenre] = useState('');
+  const [time, setTime] = useState('');
+  const [experience, setExperience] = useState('');
+  const [surveyComplete, setSurveyComplete] = useState(false);
+
+  const timeChoice = ['Full time', 'Part time', 'Temporary project']
+  const experienceChoice = ['Under education', 'Junior', 'Senior']
+
```
Comment (MarySparrow) [inline code/src/Components/Form.js:13]: Great naming here, easy to understand what its for. 👍

File: code/src/Components/Form.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault() 
+    
+    if (isSurveyComplete()) {
+      setSurveyComplete(true)
+    } else {
+      alert("Please choose options for each question")
+    }
```
Comment (MarySparrow) [inline code/src/Components/Form.js:28]: Nice that you give the reason to a user why its not submitting 👌 , always frustrating when not working.

File: code/src/Components/Form.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      <div className="circle-1"></div>
+      <p>We have noticed that you are looking for a {time} position, within the {genre} genre which accepts applications from people with {experience} experience. We will process your request.</p> 
+    </div>
+    
+  )
+ 
+} else {
+
+  return (    
+
```
Comment (MarySparrow) [inline code/src/Components/Form.js:56]: To make it same pattern as above return, remove this empty line after return💡

File: code/src/Components/Form.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+             type="radio" 
+             value={time} 
+             required
+             onChange={onTimeChange}
+             >
+           </input>
+           {time} 
+           </label>
+          ))}
+      </div>
```
Comment (MarySparrow) [inline code/src/Components/Form.js:74]: Please have a look at the indentation here, it's a little bit off. I would suggest you to move in {timeChoice.....} and maybe also {time}.  The input tag I would also suggest to move in. On line 69 close tag should go one step to the left <-- to match the input opening tag.  On line 73 I would put the (({ to match the opening of the {timeChoice...}. Am I being confusing? hope not 🙃

File: code/src/Components/Form.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+          <option>Fullstack development</option>
+
+          <option>Content writing</option>
+
+          <option>Translation</option>
+
+          <option>Customer service</option>
+
+          <option>Sales</option>
```
Comment (MarySparrow) [inline code/src/Components/Form.js:94]: Remember to keep the same pattern 🏳️‍🌈, the first option has no space between the next option. All together or all with line space will look more clean.

File: code/src/Components/Form.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+             name="experience" 
+             type="radio" 
+             value={experience} 
+             required
+             onChange={onExperienceChange}
+             >
+           </input>
+           {experience} 
+           </label>
+          ))}
```
Comment (MarySparrow) [inline code/src/Components/Form.js:114]: indentation 👀 (see above comment)

File: code/src/Components/Header.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
```
Comment (MarySparrow) [inline code/src/Components/Header.js:1]: consistency on the endings here` ; `put them on all or none . I struggle with the same cant decide 🤷‍♀️

File: code/src/index.css [hunk 8]
```diff
@@ -5,9 +5,154 @@ body {
...
 
 code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
     monospace;
 }
+
+h1 {
+  font-size: 40px;
+  margin-top: 100px;
+}
```
Comment (MarySparrow) [inline code/src/index.css:19]: in larger project its always a good practice to have classes on all styling components👩‍🏫

File: code/src/index.css [hunk 9]
```diff
@@ -5,9 +5,154 @@ body {
...
+  right: 250px;
+  height: 70px;
+  width: 70px;
+  background: cyan;
+  border-radius: 100%;
+}
+
+.last-circle {
+  position: absolute;
+  top: 30px;
```
Comment (MarySparrow) [inline code/src/index.css:50]: Really liked your circles, it brings a nice touch to the survey 🤩 🤩

File: code/src/index.css [hunk 10]
```diff
@@ -5,9 +5,154 @@ body {
...
+  flex-direction: column;
+  padding: 10px;
+  text-align: center;
+  background: lightpink;
+  height: 500px;
+  width: 400px;
+  border: 1px solid lightslategray;
+
+  margin: 30px;
+}
```
Comment (MarySparrow) [inline code/src/index.css:92]: remove empty line

File: code/src/index.css [hunk 11]
```diff
@@ -5,9 +5,154 @@ body {
...
+  align-items: center;
+  justify-content: center;
+  align-self: center;
+}
+
+.time-container {
+  padding: 10px;
+  margin: 10px;
+
+  display: flex;
```
Comment (MarySparrow) [inline code/src/index.css:105]: cleaning cleaning 🚿

File: code/src/index.css [hunk 12]
```diff
@@ -5,9 +5,154 @@ body {
...
+  flex-direction: column;
+}
+
+.footer-text {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  margin-top: 80px;
+  font-size: 20px;
```
Comment (MarySparrow) [inline code/src/index.css:157]: If you have more time to put in this project maybe add some media queries to make it responsive in mobile view.

**Review by MarySparrow — CHANGES_REQUESTED** [review]
Great job on this project Therese! Not a single warning on the accessibility validation, way to go! ✨  Be proud of what you have accomplished! 🔥🔥🔥

**Review by KarolCislowski — APPROVED** [review]
Great job!
You should read a little about how to stick your footer to the bottom of the page 🤓 
Maybe for the next step, you will try to add some more exotic types of inputs? 🍹

**Pull request closed without merging**